lib/api/filter.rb

```diff
@@ -1,5 +1,6 @@
 module Api
   class Filter
+    OPERATOR_REGEXP = /^ *([^=!>< ]+) *(!==?|[=!]~|[<>=]=?) *(.*?) *$/.freeze
```

Comment:
Can you write a spec that ensures that every key in the `OPERATORS` hash will be recognized by this regexp?  The reason I'm asking is that if someone adds a _new_ operator to the `OPERATORS` hash, we need to ensure that this Regexp also covers it.  I think the spec could look something like:

```ruby
OPERATORS.keys.each do |operator|
  it "OPERATORS_REGEXP handles the '#{operator}' operator" do
    expect("x#{operator}y").to match(OPERATORS_REGEXP)
  end
end
```

Reply:
nice! I changed to only depends upon `OPERATOR`

If the regexp is missing an entry, then `split_filter_string` will not parse it correctly.